services/dynamic-sidecar/tests/unit/test_core_rabbitmq.py

```diff
@@ -109,6 +114,22 @@ def app(mock_environment: None) -> FastAPI:
     return assemble_application()
 
 
+@pytest.fixture
+def mock_datetime_utc_now(
+    monkeypatch: MonkeyPatch,
+) -> datetime.date:
+    # Note: since datetime is unmutable, best approach is to
```

Comment:
Are you sure about this? Are you by any chance trying to ``datatime`` instead of ``datetime``?

Reply:
So I finally managed to patch the object with something inspire form https://stackoverflow.com/a/51213128/2855718
You need to wrap the class in a mock and than you can access it's methods.

But I got stopped by tenacity, which apparently uses datetime to compute some metric.

Since I don't want to spend to much time on it. this current implementation is the best I have.